api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/common/EPTSCalculationService.java

```diff
@@ -172,6 +172,31 @@ public CalculationResultMap firstEncounter(
     }
     return EptsCalculationUtils.evaluateWithReporting(def, cohort, null, null, context);
   }
+  /**
+   * Evaluates all encounters of a given type of each patient
+   *
+   * @param encounterTypes
+   * @param cohort
+   * @param location
+   * @param context
+   * @return all encounters for the patients
+   */
+  public CalculationResultMap allEncounters(
```

Comment:
So this will pull alot of data for each patient - is there a way we can filter the results a bit more? Perhaps use a date range (start-end Date) to limit how many records we'll fetch for each patient

Reply:
I have provided an end date to help limit this. But according to the requirements, it is not possible to predict the start date: **No suggestive factors of poor adherence in last 3 clinical visits**